MetaMask/metamask-extension
# Updating TextField component

## Explanation

Updating `TextField` component based on the audit and insight report
- Removing `TextFieldBase`
- Moving clear button to `TextFieldSearch`
- Renames `leftAccessory` => `startAccessory` 
- Renames `rightAccessory` => `endAccessory`
- [Insight Report](https://www.figma.com/file/aGW8sk6X6Jf9ac0MRMD4kX/TextField-Audit?node-id=282%3A20&t=xHFfsVJIbeO0FOxP-1)
* Fixes #17585 

## Screenshots/Screencaps

### Before

https://user-images.githubusercontent.com/8112138/218236107-0e69cc9b-bc59-4bc1-8e5f-08fc91a92ff5.mov

### After

https://user-images.githubusercontent.com/8112138/218236178-a461dfcc-eb55-4085-9096-12a641244457.mov

## Manual Testing Steps

- Go to the latest build of storybook in this PR
- Search for `TextFieldBase` 
- See that it doesn't exist
- Search for `TextField` in the component-library
- Check stories, controls and docs
- Check `TextFieldSearch` stories, controls and docs

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
